spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -325,13 +368,38 @@ void embed() {
 					childContext(this.app, AggregateApplicationBuilder.this.parentContext,
 							this.namespace).args(this.args).config(this.names)
 							.profiles(this.profiles).run();
-			AggregateApplicationBuilder.this.parentContext.getBeanFactory().getBean(
-					MetricsEndpoint.class).registerPublicMetrics(
-					new MetricReaderPublicMetrics(
-							new NamespaceAwareSpringIntegrationMetricReader(this.namespace, childContext.getBean(
-									IntegrationMBeanExporter.class))));
+			// Register bindable proxies as beans so they can be queried for later
+			Map<String, BindableProxyFactory> bindableProxies = BeanFactoryUtils
+					.beansOfTypeIncludingAncestors(childContext.getBeanFactory(), BindableProxyFactory.class);
+			for (String bindableProxyName : bindableProxies.keySet()) {
+				try {
+					AggregateApplicationBuilder.this.parentContext.getBeanFactory().registerSingleton(
+							this.getNamespace() + "." + bindableProxyName.substring(1, bindableProxyName.length()),
+							bindableProxies.get(bindableProxyName).getObject());
+				}
+				catch (Exception e) {
+					throw new IllegalStateException(
+							"Error while trying to register the aggregate bound interface '"
+									+ bindableProxyName + "' into namespace '" + this.getNamespace() + "'",
+							e);
+				}
+			}
+			// Register metrics if JMX enabled and exporter avalable
+			if (BeanFactoryUtils.beansOfTypeIncludingAncestors(AggregateApplicationBuilder.this.parentContext,
```

Comment:
Note: the code was just wrapped in a check, since previously it would fail if JMX was not enabled.